deploy/operator/gather.sh

```diff
@@ -111,6 +111,37 @@ function gather_imageset_data() {
   done
 }
 
+function gather_capi_data() {
+  CAPI_CRS=(agentmachine agentcluster cluster machine machinedeployment machineset nodepool hostedcluster hostedcontrolplane)
+  capi_dir="${LOGS_DEST}/capi"
+  mkdir -p "${capi_dir}"
+
+  for cr in "${CAPI_CRS[@]}"; do
+    work_dir="${capi_dir}/${cr}"
+    mkdir -p "${work_dir}"
+
+    oc_items=$(oc get ${cr} -A -o json | jq -c '.items[]') || true
+    readarray -t objects < <(echo $oc_items)
+    for obj in "${objects[@]}"; do
+      obj_name=$(echo ${obj} | jq -r .metadata.name)
+      obj_namespace=$(echo ${obj} | jq -r .metadata.namespace)
+      oc get "${cr}" -n "${obj_namespace}" "${obj_name}" -o yaml > "${work_dir}/${obj_name}_${obj_namespace}.yaml" || true
+    done
+  done
+
+  oc logs deployment/operator -n hypershift > "${capi_dir}/hypershift.log" || true
```

Comment:
We have a sample output in https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_assisted-service/3989/pull-ci-openshift-assisted-service-master-edge-e2e-metal-assisted-operator-capi/1539256049951838208/artifacts/e2e-metal-assisted-operator-capi/assisted-baremetal-operator-gather/artifacts/capi/hypershift.log and it looks sane.

We do
```
${KUBECTL} logs deployment/operator -n hypershift > ${LOGS_DEST}/hypershift/logs_operator_${DEPLOY_TARGET}.log
```
in `assisted-test-infra/scripts/download_logs.sh` so I just started with the assumption that it works there so it will work here

Reply:
Your suggestion comes as https://github.com/openshift/assisted-service/pull/4000, thanks!